core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -106,8 +106,12 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
 
     // Attempts to remove a subtopic from this topic given the
     // subtopic ID.
-    Topic.prototype.deleteSubtopic = function(subtopicId) {
+    Topic.prototype.deleteSubtopic = function(subtopicId, isNewlyCreated) {
+      var subtopicDeleted = false;
       for (var i = 0; i < this._subtopics.length; i++) {
+        if (isNewlyCreated && subtopicDeleted) {
+          this._subtopics[i].decrementId();
+        }
```

Comment:
Ditto -- probably need to decrement IDs of subtopics created after the subtopic-to-be-deleted was created?

Reply:
Only if it is newly created, would that be needed, right? Since, in the backend, such decrement is not being done for already existing subtopics in the backend.
Also, for newly created subtopics, I am already doing it.

Reply:
Yes, that's right -- it would only affect things in the case where the subtopic-to-be-deleted was newly created.

Though, would it be more robust to check based on subtopic ID, rather than position in the list? That way you don't depend on the implicit assumption that the subtopic list is sorted.